### PR TITLE
S 06 fix gallery on exposition

### PR DESCRIPTION
**Pull request by MisterPingouin**

### CODE REVIEW

**Review by Mouhamed114 — APPROVED** [review]

File: src/Controller/PictureController.php [hunk 1]
```diff
@@ -31,13 +31,27 @@ public function __construct(
...
     #[Route('/', name: 'app_picture_index', methods: ['GET'])]
-    public function index(PictureRepository $pictureRepository): Response
-    {
-        $categories = $pictureRepository->getCategories();
+    public function index(
+        PictureRepository $pictureRepository,
+        ExhibitionRepository $exhibitionRepository,
+        Request $request
+    ): Response {
+        $exhibitionId = $request->query->get('exhibition');
```
Comment (aldaitzwild) [inline src/Controller/PictureController.php]: No you should pass the id of the exhibition through the route "/{id}" and directly get an exhibition from the action parameters.

File: src/Controller/PictureController.php [hunk 2]
```diff
@@ -31,13 +31,27 @@ public function __construct(
...
+        $exhibition = null;
+
+        if ($exhibitionId) {
+            $exhibition = $exhibitionRepository->find($exhibitionId);
+        }
+
+        $pictures = $exhibition ? $pictureRepository->findBy(['exhibition' => $exhibition])
+         : $pictureRepository->findAll();
```
Comment (aldaitzwild) [inline src/Controller/PictureController.php]: No ! You won't do a findAll() if you don't the right exhibition !! If do that you'll see every picture from every exhibition even the closed ones

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

File: src/Controller/PictureController.php [hunk 3]
```diff
@@ -30,18 +30,27 @@ public function __construct(
...
-    public function index(PictureRepository $pictureRepository): Response
-    {
-        $categories = $pictureRepository->getCategories();
+    #[Route('/exhibition/{id}', name: 'app_picture_index', methods: ['GET'])]
+    public function index(
+        PictureRepository $pictureRepository,
+        ExhibitionRepository $exhibitionRepository,
+        int $id
+    ): Response {
+        $exhibition = $exhibitionRepository->find($id);
```
Comment (aldaitzwild) [inline src/Controller/PictureController.php]: No you should ask symfony to give you the exhibition thanks to the id => https://symfony.com/doc/current/doctrine.html#automatically-fetching-objects-entityvalueresolver

Comment (aldaitzwild) [reply src/Controller/PictureController.php]: Still there

**Review by aldaitzwild — CHANGES_REQUESTED** [review]

**Review by aldaitzwild — APPROVED** [review]

**Pull request merged**